src/parsers/manifest/local/parse_local_manifest.ts

```diff
@@ -50,6 +50,11 @@ export default function parseLocalManifest(
   const manifest = {
     availabilityStartTime: 0,
     duration: localManifest.duration,
+    maximumTime: {
```

Comment:
Let me reopen the debate.
Until now, the `getMaximumPosition` API of RxPlayer always returned the maximum position were you can seek in the content. That means the maximum position were the player can find an available segment, load it and append it.

Here, if we define the maximumTime (which will be used to return the maximumPosition) as the content duration, there will be probably a gap between the last segment end and the duration, if the content is not entirely downloaded.

As  a user of the API you may be confused about the fact that, if you try to seek in that gap, nothing will happen if these segments are not yet in the manifest and are not planned to be added.

Thus, if a UI decides to display for a played content the duration of the content and the current maximum seekable position (by my definition), it would be hard to know the latter if the player returns the duration as a maximum position.

Reply:
I'm also open for more debate on that one.

> That means the maximum position were the player can find an available segment, load it and append it.

No, that is your extrapolation.

My definition is just "the maximum position we can seek to" as in the "maximum position set currently for the content". I always see both the "minimumPosition" and the "maximumPosition" as the minimum and maximum bounds for the progress bar.

Some early web video players were dumb and you could either not seek until what was already downloaded or you could do it but the download would continue from the beginning which is exactly what is going on for now.

I see it also a little like a VoD content were instead of being immediately intelligent and switch to the new most wanted segment we are little more dum-dum (for now) and continue to download from the beginning. 

Also keep in mind that we've no guarantee that once the segment is in the local manifest, it will be fast to get.

For the question "how will the user know what is already downloaded ?", I answer that this is the role of the downloader to properly communicate that. I would also like to see some way to synchronize the current content playback to it. Either through the RxPlayer with another custom local-manifest api or through the application via a method in the ContentDownloader

Reply:
Thanks for going through this.

For me, it would be nice as Guillaume said to display the duration depending on how far we can seek and find a segment associated.

However, from what I understood, this informations should be provided by the **ContentDownaloader**.

Reply:
> However, from what I understood, this informations should be provided by the ContentDownaloader.

That's what the debate is for here. We're not yet sure.

My argument is that ideally, all segments in a local-manifest can be downloaded and as such the user can seek everywhere in it (I see it the same way than a regular VoD content, even if the segment scheduling logic is not the same for now).

One of the argument of @grenault73 that I understand is that the minimum and maximum positions should reflect what is actually available to seek directly.

Another thing that I would like to add is that we could perfectly handle seeks while downloading (and I hope that we do it) in the future.
In that condition:
  1. Why would that be different than for a regular VoD content?
  2. What should be the value of both properties if there are holes in the contents? E.g. considering:

      ```
      [---------|       |------|       |------------|                 ]
      0         10      15     20      30           45                60
      ```
      Should maximum position here be 10, 45 or 60?

Reply:
So, as the debate went on and on, I accept your view (as it seems to be what the majority think it should be).

But I'm still unsure of what to do in the possible case where we have non-contiguous parts.

Should the previous example return 10 or 45?